scripts/mbedtls_dev/bignum_mod_raw.py

```diff
@@ -42,6 +42,25 @@ class BignumModRawTarget(test_data_generation.BaseTarget):
 
 # BEGIN MERGE SLOT 5
 
+class BignumModRawAdd(bignum_common.ModOperationCommon,
+                      BignumModRawTarget):
+    """Test cases for bignum mpi_mod_raw_add()."""
+    symbol = "+"
+    test_function = "mpi_mod_raw_add"
+    test_name = "mbedtls_mpi_mod_raw_add"
+    input_style = "fixed"
+    arity = 2
+
+    def arguments(self) -> List[str]:
```

Comment:
Is that override needed? In bignum_common.ModOperationCommon we have

~~~~~
    def arguments(self) -> List[str]:
        return [quote_str(self.arg_n)] + super().arguments()
~~~~~

While in 

~~~~~~
    def arguments(self) -> List[str]:
        args = [quote_str(self.arg_a)]
        if self.arity == 2:
            args.append(quote_str(self.arg_b))
        return args + self.result()
~~~~~~

Which is returning the arguments in `self.arg_n, self.arg_a, self.arg_b` order.

I would recommend we refrain from overriding this method **unless it is absolutely neccessary**. If reversing the method you could achieve it in an one liner, which will not affect pylint's enforcement.

~~~~~
def arguments(self) -> List[str]:
    return list(reversed(super().arguments()))
~~~~~

Reply:
Err, I just took this from what is done in [@gabor-mezei-arm's PR (6606)](https://github.com/Mbed-TLS/mbedtls/pull/6606) :)

Reply:
I agree that we shouldn't override this unless there is no other way. I think that what you suggest @minosgalanakis is a clever solution. My first choice would be just changing the order of arguments in the test function in `test_suite_bignum_mod_raw.function`.

Reply:
(Unless of course if we have a lot of manual data as well, which of course would be a pain.)

Reply:
If I understand correctly (I've only looked superficially), we're sometimes overriding this method because different test functions take arguments in a different order? If so, overriding this is a sign that we're making things difficult for humans as well. We should keep a similar order for all test functions. Following the order of the parameters of the tested function would be my first inclination (and if those aren't consistent, we have an API design problem).

Reply:
> Following the order of the parameters of the tested function would be my first inclination

Except that in tests we pretty much uniformly put inputs before expected outputs, whereas in the new bignum modules we've kept the habi of the old bignum to put outputs first.

I won't push for changing the library function order at this point (even though I'd prefer inputs first, so instead I think we should define an order for bignum tests: first inputs in the same order as the tested function (which I think means modulus last, right?), then expected outputs, then expected result (where applicable).

Reply:
>  I think we should define an order for bignum tests: first inputs in the same order as the tested function (which I think means modulus last, right?), then expected outputs, then expected result (where applicable).

That is a good idea, however there are two PRs close to be merged and this change would impact both (conflicts/delays), so I suggest we do that in a separate PR when #6656 and #6606 are merged.